qiita_db/util.py

```diff
@@ -604,107 +617,128 @@ def insert_filepaths(filepaths, obj_id, table, filepath_table, conn_handler,
         Table that holds the filepath information
-    conn_handler : SQLConnectionHandler
-        The connection handler object connected to the DB
     move_files : bool, optional
         Whether or not to copy from the given filepaths to the db filepaths
         default: True
-    queue : str, optional
-        The queue to add this transaction to. Default return list of ids
 
     Returns
     -------
-    list or None
-        List of the filepath_id in the database for each added filepath if
-        queue not specified, or no return value if queue specified
+    list of int
+        List of the filepath_id in the database for each added filepath
     """
-    new_filepaths = filepaths
-
-    dd_id, mp = get_mountpoint(table)[0]
-    base_fp = join(get_db_files_base_dir(), mp)
-
-    if move_files:
-        # Generate the new fileapths. Format: DataId_OriginalName
-        # Keeping the original name is useful for checking if the RawData
-        # alrady exists on the DB
-        db_path = partial(join, base_fp)
-        new_filepaths = [
-            (db_path("%s_%s" % (obj_id, basename(path))), id)
-            for path, id in filepaths]
-        # Move the original files to the controlled DB directory
-        for old_fp, new_fp in zip(filepaths, new_filepaths):
-                move(old_fp[0], new_fp[0])
-
-    def str_to_id(x):
-        return (x if isinstance(x, (int, long))
-                else convert_to_id(x, "filepath_type"))
-    paths_w_checksum = [(relpath(path, base_fp), str_to_id(id),
-                        compute_checksum(path))
-                        for path, id in new_filepaths]
-    # Create the list of SQL values to add
-    values = ["('%s', %s, '%s', %s, %s)" % (scrub_data(path), pid,
-              checksum, 1, dd_id) for path, pid, checksum in
-              paths_w_checksum]
-    # Insert all the filepaths at once and get the filepath_id back
-    sql = ("INSERT INTO qiita.{0} (filepath, filepath_type_id, checksum, "
-           "checksum_algorithm_id, data_directory_id) VALUES {1} RETURNING"
-           " filepath_id".format(filepath_table, ', '.join(values)))
-    if queue is not None:
-        # Drop the sql into the given queue
-        conn_handler.add_to_queue(queue, sql, None)
-    else:
-        ids = conn_handler.execute_fetchall(sql)
-
-        # we will receive a list of lists with a single element on it
-        # (the id), transform it to a list of ids
-        return [id[0] for id in ids]
+    with TRN:
+        new_filepaths = filepaths
+
+        dd_id, mp = get_mountpoint(table)[0]
+        base_fp = join(get_db_files_base_dir(), mp)
+
+        if move_files:
+            # Generate the new fileapths. Format: DataId_OriginalName
+            # Keeping the original name is useful for checking if the RawData
+            # alrady exists on the DB
+            db_path = partial(join, base_fp)
+            new_filepaths = [
+                (db_path("%s_%s" % (obj_id, basename(path))), id)
```

Comment:
We usually avoid `id` as a var name, but I see this was like this in the original version of the code.